dataproc/dataproc.go

```diff
@@ -0,0 +1,313 @@
+	projectID   = flag.String("project", "", "Your cloud project ID.")
+	pysparkFile = flag.String("pyspark-file", "", "PySpark Job file")
+	region      = flag.String("region", "global", "Your cloud project region.")
+	zoneID      = flag.String("zone", "", "Cloud Platform zone")
+)
+
+func main() {
+	// Parse command line arguments
+	flag.Parse()
+
+	// Check for required flags (golang flags does not support?)
+	if *bucketName == "" || *clusterName == "" || *projectID == "" || *pysparkFile == "" || *region == "" || *zoneID == "" {
+		log.Fatal("Incorrect arguments specified see 'go-dataproc -help' for help")
+	}
+
+	// Create a new Dataproc service
+	service, err := getDataprocService()
+	if err != nil {
+		log.Fatal(err)
+	}
+
+	// Create a new storage service
+	storageService, err := getStorageService()
+	if err != nil {
+		log.Fatal(err)
+	}
+
+	// Create a cluster
+	_, err = createCluster(service, *clusterName, *region, *projectID, *zoneID)
+	if err != nil {
+		log.Fatal(err)
+	}
+	fmt.Println("Cluster created")
+
+	// Wait on the cluster to become active
+	fmt.Println("Waiting for cluster to be ready")
+	_, err = waitForCluster(service, *clusterName)
+	if err != nil {
+		log.Fatal(err)
+	}
+
+	// Submit a job to the cluster
+	fmt.Println("Submitting job")
+	jobId, err := submitJob(service, storageService, *pysparkFile, *projectID, *bucketName, *clusterName)
+	if err != nil {
+		log.Fatal(err)
+	}
+
+	// Wait for the job to complete
+	fmt.Println("Waiting for job to complete")
+	_, err = waitForJob(service, jobId, *projectID, *region)
+	if err != nil {
+		log.Fatal(err)
+	}
+	fmt.Println("Job is finished")
+
+	// Get the job output
+	clusterData, err := getClusterDataByName(service, *clusterName)
+	if err != nil {
+		log.Fatal(err)
+	}
+	output, err := getJobOutput(storageService, *projectID, clusterData[1], clusterData[3], jobId)
+	if err != nil {
+		log.Fatal(err)
+	}
+	fmt.Printf("Job output:\n%s\n", string(output))
+
+	// Delete the cluster
+	_, err = deleteCluster(service, *projectID, *region, *clusterName)
+	if err != nil {
+		log.Fatal(err)
+	}
+	fmt.Println("Cluster deleted")
+	fmt.Printf("Done...")
+}
+
+// getDataprocService creates and returns a Cloud Dataproc service object.
+func getDataprocService() (service *dataproc.Service, err error) {
+	ctx := context.Background()
+	client, err := google.DefaultClient(ctx, scope)
+
+	// Create a new Dataproc service
+	service, err = dataproc.New(client)
+
+	return service, err
+}
+
+// getStorageService creates and returns a Cloud Storage service object.
+func getStorageService() (service *storage.Client, err error) {
+	ctx := context.Background()
+	client, err := storage.NewClient(ctx)
+
+	return client, err
+}
+
+// createCluster creates a Cloud Dataproc cluster with the given name and region.
+func createCluster(service *dataproc.Service, name string, region string,
+	project string, zone string) (response string, err error) {
```

Comment:
Wrapping lines is a good sign that the parameters should go into a struct. These get passed around an awful lot.

Would something like this make sense?

```
type clusterConfig struct{
    project, region, zone, name string
}
```